### DISCUSSION TITLE
How to configure the delay used in dynamic import which is used to set the pastDelay boolean ?

### DISCUSSION THREAD

**Discussion by lbdremy** [vercel/next.js#48815] [Help] [answered]
### Summary

Hello,

I would like to configure the delay used to set the `pastDelay` boolean in the loading config when using dynamic import.
For example:

```
const InboxPage = dynamic(
  import('src/components/inbox/InboxPage').then((mod) => mod.InboxPage),
  {
    ssr: false,
    loading: function Spinner({ pastDelay }) {
      if (pastDelay) return <SpinnerLoader />
      return null
    },
  }
)
```

Here the spinner will be shown only when `pastDelay` is true. 
The typescript definition available for loading property:
```
(property) loading?: (({ error, isLoading, pastDelay, }: {
    error?: Error | null | undefined;
    isLoading?: boolean | undefined;
    pastDelay?: boolean | undefined;
    retry?: (() => void) | undefined;
    timedOut?: boolean | undefined;
}) => JSX.Element | null) | undefined
```
I'm not able to set the delay from the options for example at least typescript complains about it:

```
const InboxPage = dynamic(
  import('src/components/inbox/InboxPage').then((mod) => mod.InboxPage),
  {
    ssr: false,
  // line below it not possible
   delay: 2000,
    loading: function Spinner({ pastDelay }) {
      if (pastDelay) return <SpinnerLoader />
      return null
    },
  }
)
```

Here is the typescript definition for the dynamic function second argument

```
export declare type LoadableBaseOptions<P = {}> = LoadableGeneratedOptions & {
    loading?: ({ error, isLoading, pastDelay, }: {
        error?: Error | null;
        isLoading?: boolean;
        pastDelay?: boolean;
        retry?: () => void;
        timedOut?: boolean;
    }) => JSX.Element | null;
    loader?: Loader<P> | LoaderMap;
    loadableGenerated?: LoadableGeneratedOptions;
    ssr?: boolean;
```

Do you know the default delay ?
It will be cool to mention all those features in the doc here https://nextjs.org/docs/advanced-features/dynamic-import , I'm glad to contribute when all these infos are gathered.

Thanks for your help

### Additional information

```js
next version: "12.1.0"
```

**Comment by icyJoseph** [comment, marked as answer]
The `next/dynamic` module is built on top of something called Lodable, which lives here, https://github.com/vercel/next.js/blob/canary/packages/next/src/shared/lib/loadable.tsx, as the source code says:

> // Modified to be compatible with webpack 4 / Next.js 

Looking at the `next/dynamic` code, it'd seem that you just pass the delay, because it'll be passed on to the Loadable function.

You can use a `@ts-expect-error delay is not document`, kind of directive, to silence TS, and be able to test to your heart's desire.

The default delay is 200 ms, https://github.com/vercel/next.js/blob/canary/packages/next/src/shared/lib/loadable.tsx#L65

**Reply by lbdremy** [reply]
Hi @icyJoseph ,

Thanks a lot, it confirms my findings.

Have a great day
